### PR TITLE
[ticket/14562] Language fallback for extension permissions

### PR DESCRIPTION
**Pull request by kasimi**
The code was copied and adjusted from `add_mod_info()` in `includes/functions_module.php` which already has language fallback.

PHPBB3-14562

### CODE REVIEW

**Comment by Nicofuma** [comment]
You have send your PR against the develop branch which is very old.
Could you open a new PR (with the same fix) against 3.2.x please?

File: phpBB/includes/functions_admin.php [hunk 1]
```diff
@@ -3081,18 +3081,47 @@ function tidy_database()
...
+			->extension_directory('/language/en')
+			->find();
+	}
+
+	// Find files in the user's language
+	$user_lang_files = $finder
 		->prefix('permissions_')
 		->suffix(".$phpEx")
 		->core_path('language/' . $user->lang_name . '/')
 		->extension_directory('/language/' . $user->lang_name)
```
Comment (CHItA) [inline phpBB/includes/functions_admin.php]: For 3.2.x you probably achieve the same thing if you just remove the `$user->lang_name` from both places above.

Comment (kasimi) [inline phpBB/includes/functions_admin.php]: Wouldn't that load `permissions_*.php` files from *all* languages?

**Comment by kasimi** [comment]
@Nicofuma Sorry, didn't pay attention there. Against 3.2.x? Does that mean the code won't make it into 3.1.x?


Btw, the deprecated branch names are still mention [here](https://www.phpbb.com/get-involved/).

File: phpBB/includes/functions_admin.php (hunk 1 above)
Comment (CHItA) [inline phpBB/includes/functions_admin.php]: yes it would, but i don't see the problem with that.

**Comment by CHItA** [comment]
Yep, that would mean it will be <=3.2.x only. Though i'm not sure whether it has to be excluded from 3.1. @nickvergessen?

File: phpBB/includes/functions_admin.php (hunk 1 above)
Comment (kasimi) [inline phpBB/includes/functions_admin.php]: Maybe I see a problem where there is none... I think loading *all* language files would result in each language overwriting the previous one. We'd end up with always the same language being served to the user (probably the last one, alphabetically), ignoring the user-specific language preference.

Comment (CHItA) [inline phpBB/includes/functions_admin.php]: That is not really a problem, though this shouldn't be an issue for 3.2.x anyway. Did you tested it for 3.2.x or just 3.1?

Comment (kasimi) [inline phpBB/includes/functions_admin.php]: Now I have and I see why it works. I should probably submit 2 PRs, this one against 3.1.x and another one with your suggestion against 3.2.x. How should I name my branches? Or should I open another ticket?

**Comment by nickvergessen** [comment]
Well this goes against the wrong branch, develop* branches are read only...

**Comment by CHItA** [comment]
Yep, but the question is whether it should be against 3.1.x or not

**Comment by marc1706** [comment]
@kasimi Thanks for the heads up regarding the info on the get involved page. This will be updated asap.

**Comment by nickvergessen** [comment]
Looking at the state of maintenance and since this "extends functionality", I'd say 3.2+?

**Comment by kasimi** [comment]
A user is complaining because he can't understand what the extension's permissions are for. Why is displaying proper language instead of `ACL_U_SOME_PERM` a feature?

**Comment by CHItA** [comment]
i tend to agree with @kasimi.

**Pull request closed without merging**